framework/src/mesh/FileMesh.C

```diff
@@ -74,6 +82,30 @@ FileMesh::buildMesh()
     getMesh().prepare_for_use();
     getMesh().skip_partitioning(skip_partitioning_later);
   }
+  else if (_is_checkpoint)
+  {
+    CheckpointIO reader(getMesh());
+
+    reader.binary() = true;
+
+    auto & mesh = getMesh();
+
+    DistributedMesh * pmesh = cast_ptr<DistributedMesh *>(&mesh);
+
+    if (pmesh)
+      reader.parallel() = true;
+
+    reader.read(_file_name);
+
+    mesh.update_parallel_id_counts();
+    // mesh.find_neighbors(true, true);
+    // mesh.update_parallel_id_counts();
+    // mesh.redistribute();
+    // mesh.delete_remote_elements();
+    mesh.skip_partitioning(true);
```

Comment:
No, that's probably an overzealous complaint on my part.  None of our read() methods mess with partitioning.